src/riak_test_escript.erl

```diff
@@ -21,299 +21,436 @@
+-compile(export_all).
 -export([main/1]).
 -export([add_deps/1]).
 
-add_deps(Path) ->
-    {ok, Deps} = file:list_dir(Path),
-    [code:add_path(lists:append([Path, "/", Dep, "/ebin"])) || Dep <- Deps],
-    ok.
+main(Args) ->
+    %% TODO Should we use clique? -jsb
+    %% Parse command line arguments ...
+    {ParsedArgs, _NonOptionArgs} = parse_args(Args),
+    load_initial_config(ParsedArgs),
+
+    %% Configure logging ...
+    OutDir = proplists:get_value(outdir, ParsedArgs, "log"),
+    ensure_dir(OutDir),
+    lager_setup(OutDir),
+
+    {Tests, NonTests} = generate_test_lists(ParsedArgs),
+
+    ok = prepare(ParsedArgs, Tests, NonTests),
+    Results = execute(Tests, OutDir, ParsedArgs),
+    finalize(Results, ParsedArgs).
+
+% @doc Validate the command-line options
+parse_args(Args) ->
+    validate_args(getopt:parse(cli_options(), Args)).
+
+validate_args({ok, {[], _}}) ->
+    print_help();
+validate_args({ok, {ParsedArgs, NonOptionArgs}}) ->
+    case proplists:is_defined(help, ParsedArgs) of
+        true ->
+            print_help();
+        _ ->
+            {ParsedArgs, NonOptionArgs}
+    end;
+validate_args(_) ->
+    print_help().
 
-cli_options() ->
 %% Option Name, Short Code, Long Code, Argument Spec, Help Message
+cli_options() ->
 [
- {help,               $h, "help",     undefined,  "Print this usage page"},
- {config,             $c, "conf",     string,     "specifies the project configuration"},
- {tests,              $t, "tests",    string,     "specifies which tests to run"},
- {suites,             $s, "suites",   string,     "which suites to run"},
- {dir,                $d, "dir",      string,     "run all tests in the specified directory"},
- {skip,               $x, "skip",     string,     "list of tests to skip in a directory"},
- {verbose,            $v, "verbose",  undefined,  "verbose output"},
- {outdir,             $o, "outdir",   string,     "output directory"},
- {backend,            $b, "backend",  atom,       "backend to test [memory | bitcask | eleveldb]"},
- {upgrade_version,    $u, "upgrade",  atom,       "which version to upgrade from [ previous | legacy ]"},
- {keep,        undefined, "keep",     boolean,    "do not teardown cluster"},
- {report,             $r, "report",   string,     "you're reporting an official test run, provide platform info (e.g. ubuntu-1204-64)\nUse 'config' if you want to pull from ~/.riak_test.config"},
- {file,               $F, "file",     string,     "use the specified file instead of ~/.riak_test.config"}
+ {help,                   $h, "help",     undefined,  "Print this usage page"},
+ {config,                 $c, "conf",     string,     "specifies the project configuration"},
+ {tests,                  $t, "tests",    string,     "specifies which tests to run"},
+ {suites,                 $s, "suites",   string,     "which suites to run"},
+ {groups,                 $g, "groups",   string,     "specifiy a list of test groups to run"},
```

Comment:
Support for groups is not currently present in the planner.  Therefore, I think we should remove this option, and backlog completion of the feature.